src/Compilation.zig

```diff
@@ -2445,6 +2447,7 @@ fn addNonIncrementalStuffToCacheManifest(comp: *Compilation, man: *Cache.Manifes
     man.hash.add(comp.bin_file.options.z_now);
     man.hash.add(comp.bin_file.options.z_relro);
     man.hash.add(comp.bin_file.options.hash_style);
+    man.hash.add(comp.bin_file.options.compress_debug_sections orelse .none);
```

Comment:
```suggestion
    man.hash.addOptional(comp.bin_file.options.compress_debug_sections);
```

Otherwise we conflate absence of the flag with the flag carrying value `.none`.